src/credentials/FabricTable.cpp

```diff
@@ -809,4 +809,26 @@ CHIP_ERROR formatKey(FabricIndex fabricIndex, MutableCharSpan formattedKey, cons
     return err;
 }
 
+CHIP_ERROR FabricInfo::BuildFabric(ByteSpan rootCert, ByteSpan icacCert, ByteSpan nocCert, ByteSpan nodePubKey,
+                                   ByteSpan nodePrivateKey)
+{
+    Reset();
+
+    ReturnErrorOnFailure(SetRootCert(rootCert));
+    ReturnErrorOnFailure(SetICACert(icacCert));
+    ReturnErrorOnFailure(SetNOCCert(nocCert));
+
+    P256SerializedKeypair opKeysSerialized;
+    memcpy(static_cast<uint8_t *>(opKeysSerialized), nodePubKey.data(), nodePubKey.size());
+    memcpy(static_cast<uint8_t *>(opKeysSerialized) + nodePubKey.size(), nodePrivateKey.data(), nodePrivateKey.size());
+    ReturnErrorOnFailure(opKeysSerialized.SetLength(nodePubKey.size() + nodePrivateKey.size()));
+
+    P256Keypair opKey;
+    ReturnErrorOnFailure(opKey.Deserialize(opKeysSerialized));
+    ReturnErrorOnFailure(SetOperationalKeypair(&opKey));
```

Comment:
I expect this will fail with `ENABLE_HSM_CASE_OPS_KEY` but maybe that's OK if we just use this for unit tests in setups where we don't enable that.

Reply:
I'll add a not about this one